### PR TITLE
`azurerm_application_insights_workbook ` - fix update display_name without tags

### LINKED ISSUE #22113: azurerm_application_insights_workbook, cannot change workbook display_name
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.7

### AzureRM Provider Version

3.60.0

### Affected Resource(s)/Data Source(s)

azurerm_application_insights_workbook

### Terraform Configuration Files

```hcl
terraform {
  required_version = "=1.3.7"

  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "=3.60.0"
    }
  }
}

provider "azurerm" {
  features {}
}

# example from documentation
# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/application_insights_workbook
resource "azurerm_application_insights_workbook" "main" {
  name                = "96c11918-506e-4bae-974b-5f2dc44398f7"

  resource_group_name = local.resource_group_name
  location            = local.resource_group_location
  
  display_name        = "workbook1"
  
  data_json = jsonencode({
    "version" = "Notebook/1.0",
    "items" = [
      {
        "type" = 1,
        "content" = {
          "json" = "Test2022"
        },
        "name" = "text - 0"
      }
    ],
    "isLocked" = false,
    "fallbackResourceIds" = [
      "Azure Monitor"
    ]
  })

  tags = {
    ENV = "Test"
  }
}
```

### Debug Output/Panic Output

```shell
n/a
```

### Expected Behaviour

When display_name is changed, terraform apply should NOT fail.

### Actual Behaviour

When display_name is changed, terraform apply FAILS.

Plan
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # azurerm_application_insights_workbook.main will be updated in-place
  ~ resource "azurerm_application_insights_workbook" "main" {
      ~ display_name        = "workbook1" -> "workbook1-chage"
        id                  = "REDACRED"
        name                = "96c11918-506e-4bae-974b-5f2dc44398f7"
        tags                = {
            "ENV" = "Test"
        }
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

Actual apply
```
│ Error: updating Workbook (Subscription: "REDACRED"
│ Resource Group Name: "REDACRED"
│ Workbook Name: "96c11918-506e-4bae-974b-5f2dc44398f7"): unexpected status 400 with error: BadRequest: Invalid hidden-title tag: 'workbook1', expecting hidden-title to be the same as properties.displayName: 'workbook1-chage'
│
│   with azurerm_application_insights_workbook.main,
│   on main.tf line 8, in resource "azurerm_application_insights_workbook" "main":
│    8: resource "azurerm_application_insights_workbook" "main" {
│
│ updating Workbook (Subscription: "REDACRED"
│ Resource Group Name: "REDACRED"
│ Workbook Name: "96c11918-506e-4bae-974b-5f2dc44398f7"): unexpected status 400 with error: BadRequest: Invalid hidden-title tag: 'workbook1', expecting hidden-title to be the same as properties.displayName:
│ 'workbook1-chage
```

Trying to change `hidden-title` tags via tags
```

...

   display_name        = "workbook1-chage"
...  
  tags = {
    ENV = "Test",
    "hidden-title" = "workbook1-chage",
  }

# plan fails
Error: a tag with the key `hidden-title` should not be used to set the display name. Please Use `display_name` instead

```

Changing display_name tag:
```
   display_name        = "workbook1-chage"
...  
  tags = {
    ENV = "Test",
    "display_name" = "workbook1-chage",
  }
```

Plan
```
  # azurerm_application_insights_workbook.main will be updated in-place
  ~ resource "azurerm_application_insights_workbook" "main" {
      ~ display_name        = "workbook1" -> "workbook1-chage"
        id                  = "...."
        name                = "96c11918-506e-4bae-974b-5f2dc44398f7"
      ~ tags                = {
          + "display_name" = "workbook1-chage"
            # (1 unchanged element hidden)
        }
        # (5 unchanged attributes hidden)
    }
```

Plan/Apply is OK

Changing both display name and tag together
```
  # azurerm_application_insights_workbook.main will be updated in-place
  ~ resource "azurerm_application_insights_workbook" "main" {
      ~ display_name        = "workbook1-chage" -> "workbook1-chage-2"
        id                  = "..."
        name                = "96c11918-506e-4bae-974b-5f2dc44398f7"
      ~ tags                = {
          ~ "display_name" = "workbook1-chage" -> "workbook1-chage-2"
            # (1 unchanged element hidden)
        }
        # (5 unchanged attributes hidden)
    }
```

Plan/Apply is OK

### Steps to Reproduce

* terraform apply
* change display_name 
* terraform apply - this leads to apply failure, TF cannot change tags

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by teowa**
- resolves #22113 

```
TF_ACC=1 go test -v ./internal/services/applicationinsights -parallel 50 -test.run=TestAccApplicationInsightsWorkbook_ -timeout 1440m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccApplicationInsightsWorkbook_basic
=== PAUSE TestAccApplicationInsightsWorkbook_basic
=== RUN   TestAccApplicationInsightsWorkbook_requiresImport
=== PAUSE TestAccApplicationInsightsWorkbook_requiresImport
=== RUN   TestAccApplicationInsightsWorkbook_complete
=== PAUSE TestAccApplicationInsightsWorkbook_complete
=== RUN   TestAccApplicationInsightsWorkbook_updateDisplanName
=== PAUSE TestAccApplicationInsightsWorkbook_updateDisplanName
=== RUN   TestAccApplicationInsightsWorkbook_update
=== PAUSE TestAccApplicationInsightsWorkbook_update
=== RUN   TestAccApplicationInsightsWorkbook_hiddenTitleInTags
=== PAUSE TestAccApplicationInsightsWorkbook_hiddenTitleInTags
=== CONT  TestAccApplicationInsightsWorkbook_basic
=== CONT  TestAccApplicationInsightsWorkbook_updateDisplanName
=== CONT  TestAccApplicationInsightsWorkbook_complete
=== CONT  TestAccApplicationInsightsWorkbook_hiddenTitleInTags
=== CONT  TestAccApplicationInsightsWorkbook_update
=== CONT  TestAccApplicationInsightsWorkbook_requiresImport
--- PASS: TestAccApplicationInsightsWorkbook_hiddenTitleInTags (6.70s)
--- PASS: TestAccApplicationInsightsWorkbook_basic (133.02s)
--- PASS: TestAccApplicationInsightsWorkbook_requiresImport (137.27s)
--- PASS: TestAccApplicationInsightsWorkbook_complete (156.84s)
--- PASS: TestAccApplicationInsightsWorkbook_updateDisplanName (160.42s)
--- PASS: TestAccApplicationInsightsWorkbook_update (273.28s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/applicationinsights   276.837s

```

### CODE REVIEW

File: internal/services/applicationinsights/application_insights_workbook_resource_test.go [hunk 1]
```diff
@@ -59,6 +59,27 @@ func TestAccApplicationInsightsWorkbook_complete(t *testing.T) {
 	})
 }
 
+func TestAccApplicationInsightsWorkbook_updateDisplanName(t *testing.T) {
```
Comment (catriona-m) [inline internal/services/applicationinsights/application_insights_workbook_resource_test.go:62]:
```suggestion
func TestAccApplicationInsightsWorkbook_updateDisplayName(t *testing.T) {
```

File: internal/services/applicationinsights/application_insights_workbook_resource.go [hunk 2]
```diff
@@ -217,6 +217,7 @@ func (r ApplicationInsightsWorkbookResource) Update() sdk.ResourceFunc {
 
 			if metadata.ResourceData.HasChange("display_name") {
 				properties.Properties.DisplayName = model.DisplayName
+				delete(*properties.Tags, "hidden-title")
```
Comment (catriona-m) [inline internal/services/applicationinsights/application_insights_workbook_resource.go:220]: do we need to check for nil here before dereferencing `*properties.Tags`?

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks @teowa - I have left a couple of minor comments inline here but can take another look once those get fixed up. Thanks!

File: internal/services/applicationinsights/application_insights_workbook_resource_test.go (hunk 1 above)
Comment (teowa) [reply internal/services/applicationinsights/application_insights_workbook_resource_test.go:62]: Thanks, I have changed the code.

**Pull request merged**